recipe-server/client/control_new/components/common/BaseAntForm.js

```diff
@@ -0,0 +1,94 @@
+import autobind from 'autobind-decorator';
+import { Form, message } from 'antd';
+import React, { PropTypes as pt } from 'react';
+
+
+/**
+ * Base class for antd-based forms that handles displaying errors
+ * from the API.
+ */
+export default class BaseAntForm extends React.Component {
+  static propTypes = {
+    form: pt.object.isRequired,
+    onSubmit: pt.func.isRequired,
+    errors: pt.object,
+  }
+
+  @autobind
+  handleSubmit(event) {
+    event.preventDefault();
+    this.triggerSubmit();
+  }
+
+  @autobind
+  async triggerSubmit() {
+    try {
+      const values = await this.validateFields();
+      this.props.onSubmit(values);
+    } catch (error) {
+      message.error('Could not validate form. Please correct the errors below.');
+    }
+  }
+
+  @autobind
+  validateFields() {
```

Comment:
We still need the Promise for `form.validateFields`, right? This is just to make it clear it's effectively async? Just want to make sure I'm not missing if the Promise isn't necessary.